src/containers/cart-header/cart-header.js

```diff
@@ -1,32 +1,34 @@
-import React, { useMemo } from 'react';
+import React, { useEffect, useMemo } from 'react';
 import { Link } from 'react-router-dom';
-import { useSelector } from 'react-redux';
-
+import { useDispatch, useSelector } from 'react-redux';
 import Badge from '@material-ui/core/Badge';
 import IconButton from '@material-ui/core/IconButton';
 import CartIcon from './CartIcon';
-
 import { useStyles } from './cart-header.styles';
+import { setCart } from '../../redux/common-store/common.actions';
+import { getFromLocalStorage } from '../../services/local-storage.service';
+import { CART_KEY } from '../../configs';
 import routes from '../../configs/routes';
-import { cartKey } from '../../configs';
 
 const { pathToCart } = routes;
 
 const CartHeader = ({ fromSideBar }) => {
-  const { cartItems } = useSelector(({ CommonStore }) => ({
-    cartItems: CommonStore.cart
-  }));
   const styles = useStyles({ fromSideBar });
+  const dispatch = useDispatch();
+  const cartItems = useSelector(({ CommonStore }) => CommonStore.cart);
   const itemsCount = useMemo(
     () => cartItems.length && cartItems.reduce((acc, item) => acc + item.quantity, 0),
     [cartItems]
   );
+  useEffect(() => {
+    dispatch(setCart(getFromLocalStorage(CART_KEY)));
```

Comment:
we have tried to avoid using redux. Use useCart hook for manage cart

Reply:
Without using the redux store I cannot get the badge updates from other components. I will try to find out if there is some good solution to this issue.

Reply:
Rewrote it using the React Context.